impl/src/planning.rs

```diff
@@ -47,251 +45,157 @@ pub trait BuildPlanner {
    * A function that returns a completely planned build using internally generated metadata, along
    * with settings, platform specifications, and critical file locations.
    */
-  fn plan_build(
-    &mut self,
-    settings: &RazeSettings,
-    path_prefix: &PathBuf,
-    files: CargoWorkspaceFiles,
-    platform_details: Option<PlatformDetails>,
-  ) -> Result<PlannedBuild>;
+  fn plan_build(&self, platform_details: Option<PlatformDetails>) -> Result<PlannedBuild>;
 }
 
 /** The default implementation of a `BuildPlanner`. */
-pub struct BuildPlannerImpl<'fetcher> {
-  metadata_fetcher: &'fetcher mut dyn MetadataFetcher,
-  binary_deps_tempdir: Result<TempDir, io::Error>,
+pub struct BuildPlannerImpl {
+  pub metadata: RazeMetadata,
+  pub settings: RazeSettings,
```

Comment:
Why is this stuff now pub?

Reply:
(i'm making this comment not because I have deep wisdom about this, but because it looks superficially like you might be breaking the abstraction to take an implementation shortcut)

Reply:
It didn't need to be. Updated

Reply:
This still looks pub as of most recent commit.

Reply:
Sorry 😢 I did forget to add that file.